subsys/bluetooth/controller/ll_sw/nordic/lll/lll_central_iso.c

```diff
@@ -4,36 +4,913 @@
+
+	} else {
+		uint32_t ret;
+
+		ret = lll_prepare_done(cig_lll);
+		LL_ASSERT(!ret);
+	}
+
+	DEBUG_RADIO_START_A(1);
+
+	return 0;
+}
+
+static void isr_tx(void *param)
+{
+	struct lll_conn_iso_stream *cis_lll;
+	struct node_rx_pdu *node_rx;
+	struct lll_conn *conn_lll;
+	uint32_t hcto;
+
+	/* Clear radio tx status and events */
+	lll_isr_tx_status_reset();
+
+	/* Close subevent, one tx-rx chain */
+	radio_switch_complete_and_disable();
+
+	/* Get reference to CIS LLL context */
+	cis_lll = param;
+
+	/* Get reference to ACL context */
+	conn_lll = ull_conn_lll_get(cis_lll->acl_handle);
+
+	/* Acquire rx node for reception */
+	node_rx = ull_iso_pdu_rx_alloc_peek(1U);
+	LL_ASSERT(node_rx);
+
+	/* Encryption */
+	if (false) {
+
+#if defined(CONFIG_BT_CTLR_LE_ENC)
+	} else if (conn_lll->enc_rx) {
+		uint64_t payload_count;
+		uint8_t pkt_flags;
+
+		payload_count = (cis_lll->event_count * cis_lll->rx.bn) +
+				(bn_rx - 1U);
+		cis_lll->rx.ccm.counter = payload_count;
+
+		pkt_flags = RADIO_PKT_CONF_FLAGS(RADIO_PKT_CONF_PDU_TYPE_DC,
+						 cis_lll->rx.phy,
+						 RADIO_PKT_CONF_CTE_DISABLED);
+		radio_pkt_configure(RADIO_PKT_CONF_LENGTH_8BIT,
+				    (cis_lll->rx.max_pdu + PDU_MIC_SIZE),
+				    pkt_flags);
+		radio_pkt_rx_set(radio_ccm_rx_pkt_set(&cis_lll->rx.ccm,
+						      cis_lll->rx.phy,
+						      node_rx->pdu));
+#endif /* CONFIG_BT_CTLR_LE_ENC */
+
+	} else {
+		uint8_t pkt_flags;
+
+		pkt_flags = RADIO_PKT_CONF_FLAGS(RADIO_PKT_CONF_PDU_TYPE_DC,
+						 cis_lll->rx.phy,
+						 RADIO_PKT_CONF_CTE_DISABLED);
+		radio_pkt_configure(RADIO_PKT_CONF_LENGTH_8BIT,
+				    cis_lll->rx.max_pdu, pkt_flags);
+		radio_pkt_rx_set(node_rx->pdu);
+	}
+
+	/* assert if radio packet ptr is not set and radio started rx */
+	LL_ASSERT(!radio_is_ready());
+
+	/* +/- 2us active clock jitter, +1 us hcto compensation */
+	hcto = radio_tmr_tifs_base_get() + EVENT_IFS_US +
+	       (EVENT_CLOCK_JITTER_US << 1) + RANGE_DELAY_US +
+	       HCTO_START_DELAY_US;
+#if defined(CONFIG_BT_CTLR_PHY)
+	hcto += radio_rx_chain_delay_get(cis_lll->rx.phy, PHY_FLAGS_S8);
+	hcto += addr_us_get(cis_lll->rx.phy);
+	hcto -= radio_tx_chain_delay_get(cis_lll->tx.phy,
+					 cis_lll->tx.phy_flags);
+#else /* !CONFIG_BT_CTLR_PHY */
+	hcto += radio_rx_chain_delay_get(0U, 0U);
+	hcto += addr_us_get(0U);
+	hcto -= radio_tx_chain_delay_get(0U, 0U);
+#endif /* !CONFIG_BT_CTLR_PHY */
+
+	radio_tmr_hcto_configure(hcto);
+
+#if defined(CONFIG_BT_CTLR_PROFILE_ISR) || \
+	defined(HAL_RADIO_GPIO_HAVE_PA_PIN)
+	radio_tmr_end_capture();
+#endif /* CONFIG_BT_CTLR_PROFILE_ISR */
+
+#if defined(HAL_RADIO_GPIO_HAVE_LNA_PIN)
+	radio_gpio_lna_setup();
+
+#if defined(CONFIG_BT_CTLR_PHY)
+	radio_gpio_pa_lna_enable(radio_tmr_tifs_base_get() + EVENT_IFS_US - 4 -
+				 radio_tx_chain_delay_get(cis_lll->tx.phy,
+							  cis_lll->tx.phy_flags) -
+				 HAL_RADIO_GPIO_LNA_OFFSET);
+#else /* !CONFIG_BT_CTLR_PHY */
+	radio_gpio_pa_lna_enable(radio_tmr_tifs_base_get() + EVENT_IFS_US - 4 -
```

Comment:
What is the `-4` here? Do we have a `define` for those?

Reply:
Updated